src/lib/xmltok_impl.rs

```diff
@@ -1565,17 +1566,16 @@ impl<T: XmlEncodingImpl+XmlTokImpl> XmlEncoding for T {
     unsafe fn getAtts(
         &self,
         mut buf: ExpatBufRef,
-        mut attsMax: libc::c_int,
-        mut atts: *mut ATTRIBUTE,
-    ) -> libc::c_int {
+        f: &mut dyn FnMut(ATTRIBUTE) -> XML_Error,
+    ) -> XML_Error {
         #[derive(PartialEq)]
         enum State {
             Other,
             InName,
             InValue,
         };
         let mut state = State::InName;
-        let mut nAtts: libc::c_int = 0 as libc::c_int;
+        let mut att: ATTRIBUTE = std::mem::zeroed();
```

Comment:
We should probably implement `Default` for `ATTRIBUTE` and use that here instead.

Reply:
Going to do that, since `zeroed` is unsafe anyway. If we remove that and 3 calls to `offset`, we can remove unsafe from `getAtts` altogether.

Reply:
I decided to just build the `ATTRIBUTE` object manually inside `getAtts`, since that's the only place it's used anyway.